packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -206,6 +189,39 @@ export class Arn {
     });
   }
 
+  /**
+   * Extract the full resource name from an ARN
+   *
+   * Necessary for resource names (paths) that may contain the separator, like
+   * `arn:aws:iam::111111111111:role/path/to/role/name`.
+   *
+   * Only works if we statically know the expected `resourceType` beforehand, since we're going
+   * to use that to split the string on ':<resourceType>/' (and take the right-hand side).
+   *
+   * We can't extract the 'resourceType' from the ARN at hand, because CloudFormation Expressions
+   * only allow literals in the 'separator' argument to `{ Fn::Split }`, and so it can't be
+   * `{ Fn::Select: [5, { Fn::Split: [':', ARN] }}`.
+   *
+   * Only necessary for ARN formats for which the type-name separator is `/`.
+   */
+  public static extractResourceName(arn: string, resourceType: string): string {
```

Comment:
Make the 2nd parameter a `ExtractResourceNameOptions` so it's extensible.

Reply:
I don't really see the need for that. We have multiple extension points already:

* Optional 3rd parameter which can be a struct
* Additional method

No need for premature generalization.

Reply:
I see the 2nd parameter is not optional, so ok.